src/java.base/share/classes/jdk/internal/classfile/impl/verifier/VerifierImpl.java

```diff
@@ -114,9 +114,10 @@ public static List<VerifyError> verify(ClassModel classModel, Consumer<String> l
     }
 
     public static List<VerifyError> verify(ClassModel classModel, ClassHierarchyResolver classHierarchyResolver, Consumer<String> logger) {
-        var klass = new VerificationWrapper(classModel);
-        log_info(logger, "Start class verification for: %s", klass.thisClassName());
+        String clsName = classModel.thisClass().asInternalName();
```

Comment:
This can still throw `ConstantPoolException` if this_class points to a non-Class entry. This entry is lazily read by `ClassReader`, so you can create a `ClassModel` for such a bad class.

Reply:
Alternatively, a malformed Class constant can point to a non-utf8, so the `asInternalName` can fail too.

Reply:
Right, it will still throw if you pass a broken model to `ClassFile::verify(ClassModel)`.
I'll catch this top-level exception in both `ClassFile::verify` methods.